### PR TITLE
[XLA:CPU][oneDNN] Enable oneDNN MatMul + BiasAdd + Sigmoid fusion.

### PR DESCRIPTION
**Pull request by nhatleSummer22**
This PR enables oneDNN MatMul+BiasAdd+Sigmoid fusion for FP32, BF16 and F16.

### CODE REVIEW

**Comment by thomasjoerg** [comment]
Hi @sgerrard , do you have time to take a look or want me to find somebody else?

**Comment by sgerrard** [comment]
@penpornk should review this, I've requested a review from her.

File: xla/tests/onednn_matmul_test.cc [hunk 1]
```diff
@@ -1062,6 +1070,94 @@ TEST_F(MatmulTest, BiasAddELUFusion_F16_2) {
   MatchOptimizedHlo(matmul_module_str, fused_matmul_bias_elu_rewrite_str_);
 }
 
+TEST_F(MatmulTest, SIGMOIDTestF32) {
```
Comment (jreiffers) [inline xla/tests/onednn_matmul_test.cc:1073]: Mega nit: `Sigmoid`?

File: xla/service/cpu/onednn_matmul_rewriter.cc [hunk 2]
```diff
@@ -762,6 +762,30 @@ class OneDnnMatMulRewriteVisitor : public DfsHloRewriteVisitor {
...
+  Status HandleDivide(HloInstruction* instr) override {
+    HloInstruction* matmul_call;
+    HloInstruction* intermediate_instr = nullptr;
+    HloInstruction* optional_bitcast = nullptr;
+    HloInstruction* src;
+    if (SigmoidActivation(instr, &src)) {
+      if (Match(src, ElementwiseSafeIntermediates(
+                         &intermediate_instr, &optional_bitcast,
+                         OneDnnMatmulInstr(&matmul_call))
+                         .WithOneUser())) {
```
Comment (jreiffers) [inline xla/service/cpu/onednn_matmul_rewriter.cc:781]: Would it maybe be clearer if this was in the Activation function? `m::Op(src).WithOneUser()`? Or if it was just `if (src->user_count() == 1 && Match(src, ...))`? No strong opinions either way.

**Review by jreiffers — APPROVED** [review]

**Pull request closed without merging**